src/frontends/pytorch/src/op/rot90.cpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/frontend/pytorch/node_context.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/transpose.hpp"
+#include "utils.hpp"
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+using namespace ov::op;
+
+OutputVector translate_rot90(const NodeContext& context) {
+    num_inputs_check(context, 1, 3);
+    auto input = context.get_input(0);
+    int k = context.input_is_none(1) ? 1 : context.const_input<int64_t>(1);
+    std::vector<int64_t> dims = context.input_is_none(2) ? std::vector<int64_t>{0, 1}
+                                                         : context.const_input<std::vector<int64_t>>(2);
+    const auto& partial_shape = input.get_partial_shape();
+    const auto ndims = partial_shape.rank().get_length();
+
+    PYTORCH_OP_CONVERSION_CHECK(dims.size() == 2,
+                                "Expected total rotation dims == 2, but got dims = ",
+                                dims.size());
+    PYTORCH_OP_CONVERSION_CHECK(ndims >= 2,
+                                "Expected total dims >= 2, but got total dims = ",
+                                ndims);
+    PYTORCH_OP_CONVERSION_CHECK(dims[0] != dims[1],
+                                "Rotation dimensions must be different, but got dim0 = " +
+                                    std::to_string(dims[0]) + " and dim1 = " + std::to_string(dims[1]));
+
+    for (auto& dim : dims) {
+        dim = (dim + ndims) % ndims;
+    }
```

Comment:
Use `normalize_axis` function for this after you change `dims` to be a Node rather then constant.